### ISSUE TITLE
Support for unsubscribe groups API

### ISSUE THREAD

**Issue by fernandosesma** [janjaali/sendGrid-mock#71]
Hello! My team could have use for this project to do contract testing. One feature missing for us is support for the groups APIs (`/v3/asm/groups`). Would you be willing to look at a PR for the addition of these APIs to your mock server project? Thanks!

**Comment by janjaali** [comment]
Sure, we can try that and thx for your contribution in advance! Most likely you're interested in integrating only a subset of the `/v3/asm/groups` API into this project, correct?

**Comment by fernandosesma** [comment]
Of course! And yes specifically the _unsubscribe groups_ APIs. As in the following: 
![image](https://github.com/janjaali/sendGrid-mock/assets/36521828/d748e113-c1d0-4b89-9705-cd0bcca04230)
